src/plugins/enzymes/src/EnzymesIO.cpp

```diff
@@ -72,25 +72,51 @@ QList<SEnzymeData> EnzymesIO::readEnzymes(const QString& url, U2OpStatus& os) {
     }
 
     QList<SEnzymeData> resToDelete;
-    // assign alphabet if needed.
+    // do additional calculations to recognize some enzymes features
     for (int i = 0, n = res.count(); i < n; i++) {
         SEnzymeData& d = res[i];
         if (d->seq == QByteArray("?")) {
             algoLog.trace(tr("The enzyme '%1' has unknown sequence").arg(d->id));
             resToDelete.append(d);
         } else {
+            // calculate enzyme alphabet
+            d->alphabet = U2AlphabetUtils::findBestAlphabet(d->seq);
             if (d->alphabet == nullptr) {
-                d->alphabet = U2AlphabetUtils::findBestAlphabet(d->seq);
-                if (!d->alphabet) {
-                    algoLog.error(tr("No enzyme alphabet: '%1', sequence '%2'")
-                                      .arg(d->id)
-                                      .arg(QString(d->seq)));
+                algoLog.error(tr("No enzyme alphabet: '%1', sequence '%2'")
+                                    .arg(d->id)
+                                    .arg(QString(d->seq)));
+            } else if (!d->alphabet->isNucleic()) {
+                algoLog.error(tr("Non-nucleic enzyme alphabet: '%1', alphabet: %2, sequence '%3'")
+                                    .arg(d->id)
+                                    .arg(d->alphabet->getId())
+                                    .arg(QString(d->seq)));
+            }
+            // calculate enzyme overhang types
+            auto seqSize = d->seq.size();
+            if (d->cutDirect == ENZYME_CUT_UNKNOWN) {
+                d->overhangTypes |= EnzymeData::OverhangType::NoOverhang;
+            } else {
+                if (qFuzzyCompare((double)seqSize / 2, d->cutDirect) && d->cutDirect == d->cutComplement) {
```

Comment:
> Why qFuzzyCompare with int? Can you just cast the int into double to avoid a warning?

`int` explicitly casts to "double", isn't it? And I do not get the idea about warning - I do not have any warnings on WIndows.

> nit: put the result of the comparison into a named variable - the algorithm will be more readable.
OK

Reply:
Skip the question about compiler warnings - the reason I asked is a non-standard code to deal with integers using fuzzy-compare for doubles.

Looks like the current algorithm has a problem. Consider the following scenario:

1. Direct cut is in the middle of the sequence: a, b, c, d, e 
2. length = 5, directCutIndex = 2 (c). 
3. Your algorithm: length / 2 = 2.5, so directCut is not in the middle.

Correct check (no fuzzy & doubles): 
`bool directCutInTheMiddleOfSequence  = (length % 2 ==1) && ((length - 1) / 2) == directCut;`

wdyt?

Reply:
A bit incorrect statement, but OK, fixed.

Reply:
I see so the directCutInTheMiddleOfSequence should point to a position between 2 bases, but not on the middle base itself.